### PR TITLE
packaging management components, clusterclass-based creation

### LINKED ISSUE #1531: Create a cluster using a clusterclass-based configuration file
**Describe the feature request**

As a user I would like to create a workload cluster using a configuration file with a definition of a CAPI Cluster resource referencing a ClusterClass appropriate for target AWS infrastructure.

When provided in this form, the input does not require any transformation. This task is more about updating the validation logic to account for this format, taking advantage of any early-fail opportunities, as well as ensuring that the logic to monitor the creation process continues to work regardless of whether the cluster creation is CC-based or not.

**Describe alternatives you've considered**
NA

**Affected product area**

CLI
Cluster Lifecycle

Additional context

### PR DESCRIPTION
**Pull request by vuil**
### What this PR does / why we need it

Changes pertaining to packaging management components and enabling cluster creation using a clusterclass based Cluster resource definition (CCluster.yaml)
Packaged and ClusterClass based cluster LCM
and some github workflow changes

### Which issue(s) this PR fixes

Related-to #1339
Related-to #1340
Related-to #1341
Fixes #1531

### Describe testing done for PR

```
export OCI_REGISTRY=...
export COMPONENTS="pkg/v2/tkr/controller/tkr-source pkg/v2/tkr/controller/tkr-status pkg/v1/sdk/features addons pkg/v2/tkr/webhook/infra-machine pkg/v1/sdk/capabilities pkg/v2/tkr/webhook/tkr-conversion pkg/v2/tkr/webhook/cluster/tkr-resolver addons/pinniped/tanzu-auth-controller-manager"
export BUILD_VERSION="v0.21.0"

make docker-build
make docker-publish
make local-registry
make kbld-image-replace
make package-push-bundles-repo PACKAGE_REPOSITORY=management
make clean-registry
```

### Release note

```release-note
TBA
```

### PR Checklist

- x Squash the commits into one or a small number of logical commits
- x Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- x Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: hack/kind/Testing_Packages.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Testing packages
```
Comment (vuil) [inline hack/kind/Testing_Packages.md:1]: @anujc25 which of the 3 files in this directory are worth retaining?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Main
 
 on:
   pull_request:
-    branches: [ main, release-* ]
+    branches: [ main, release-*, package-based-lcm ]
```
Comment (vuil) [inline .github/workflows/main.yaml:5]: perhaps we should remove references to this branch before we merge to main?

File: hack/kind/Testing_Packages.md (hunk 1 above)
Comment (anujc25) [reply hack/kind/Testing_Packages.md:1]: `deploy_kind_with_capi_and_kapp.sh` file depends on CAPx versions. If we are keeping this we might have to periodically update the versions there.
Other than that, I thing we can keep all 3 files here.

**Comment by alfredthenarwhal** [comment]
> @vuil:  /test install-vc7
Commit: fe4594f

Build failed! Build no: 2156

**Comment by alfredthenarwhal** [comment]
> @vuil:  /test install-vc7
Commit: e37bae2

Tests failed! Build no: 2167

**Comment by alfredthenarwhal** [comment]
> @vuil:  /test install-vc7
Commit: 9e13af4

Tests failed! Build no: 2168

**Comment by alfredthenarwhal** [comment]
> @vuil:  /test install-vc7
Commit: 9e13af4

Tests failed! Build no: 2169

**Comment by navidshaikh** [comment]
> @vuil: /test install-vc7
> Commit: cdd2e07

Tests passed! Build no: 2170

(CI signal failed because of a network flake)

File: .github/workflows/tkg_integration_tests.yaml [hunk 3]
```diff
@@ -185,3 +185,120 @@ jobs:
         done
       env:
         cluster_prefix: ${{ env.CLUSTER_PREFIX }}
+  integ-tests-aws-cc:
```
Comment (vuil) [inline .github/workflows/tkg_integration_tests.yaml:188]: note: will move this workflow change out, into a subsequent change.

**Review by anujc25 — APPROVED** [review]
Looks good

File: .github/workflows/main.yaml (hunk 2 above)
Comment (vuil) [reply .github/workflows/main.yaml:5]: will cleanup workflow changes in a subsequent change.

**Pull request merged**